backend/internal/controllers/book.go

```diff
@@ -209,3 +210,62 @@ func (c *BookController) UpdateBookDetails(rw http.ResponseWriter, req *http.Req
 	writeResponse(rw, http.StatusOK, resBookDetails)
 
 }
+
+// increments a book's click count for the current day (/analytics/{id}/inc)
+func (c *BookController) UpdateBookClicks(rw http.ResponseWriter, req *http.Request) {
+	var bookID string = chi.URLParam(req, "id")
+
+	analytics, _ := c.Books.FetchBookAnalytics(req.Context(), bookID, 1)
+
+	if analytics == nil {
+		writeResponse(rw, http.StatusNotFound, "Book analytics not found for requested id")
+		return
+	}
+
+	err := c.Books.UpdateBookAnalytics(req.Context(), bookID)
+
+	if err != nil {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		return
+	}
+
+	writeResponse(rw, http.StatusNoContent, nil)
```

Comment:
Oh, also, I think StatusOK makes more sense here than NoContent - you don't always have to return 204 if there's nothing in the body, I've really only seen it as a convention for DELETE routes.